tensorflow/python/keras/layers/recurrent_v2.py

```diff
@@ -1333,6 +1336,36 @@ def cudnn_lstm(inputs, init_h, init_c, kernel, recurrent_kernel, bias, mask,
   # so that mathematically it is same as the canonical LSTM implementation.
   full_bias = array_ops.concat((array_ops.zeros_like(bias), bias), 0)
 
+  if build_info.is_rocm_build:
+    units = int(int(kernel.get_shape()[1]) / 4)
+    # ROCm MIOpen's weight sequence for LSTM is different from both canonical
+    # and Cudnn format
+    # MIOpen: [i, f, o, c] Cudnn/Canonical: [i, f, c, o]
+    # i is input gate weights.
+    # f is forget gate weights.
+    # o is output gate weights.
+    # c is cell gate weights.
+    weights=[
```

Comment:
Please use tf.split with 4 to rearrange the weights, rather than sliding it. The split approach is much performant.